### ISSUE TITLE
Versions 0.2.0 and newer depend on `bash`, don't work in official Hashicorp Terraform containers

### ISSUE THREAD

**Issue by jantari** [Invicton-Labs/terraform-external-shell-resource#4]
Hello,

I was going to verify that #1 is fixed, but we ran into a blocking issue with all versions after 0.1.2

We run our terraform steps inside the [official `hashicorp/terraform` docker containers](https://hub.docker.com/r/hashicorp/terraform), which I suspect is quite common.

It appears that the more recent versions of the module cannot work with environments that don't have `bash`, such as the commonly used Alpine Linux. This is what we get:

```
module.il_module[0].null_resource.shell (local-exec): Executing: ["/bin/sh" "-c" "bash /builds/platforms/repositoryname/environments/prod/.terraform/modules/il_module/run.sh L2J1aWxkcy9wbGF0Zm9ybXMvcmVwb3NpdG9yeW5hbWUvZW52aXJvbm1lbnRzL3Byb2QvLnRlcnJhZm9ybS9tb2R1bGVzL2lsX21vZHVsZS90bXBmaWxlcw== 19816ec5-e5b8-d70a-b9e0-7c3ae0af6e12 1 0 1 YmFzZTY0LWVuY29kZWQtcmVkYWN0ZWQtY3VybC1jb21tYW5k 19816ec5-e5b8-d70a-b9e0-7c3ae0af6e12.stdout 19816ec5-e5b8-d70a-b9e0-7c3ae0af6e12.stderr 19816ec5-e5b8-d70a-b9e0-7c3ae0af6e12.exitcode 0 0 0"]
module.il_module[0].null_resource.shell (local-exec): /bin/sh: bash: not found

╷
│ Error: local-exec provisioner error
│ 
│   with module.il_module[0].null_resource.shell,
│   on .terraform/modules/il_module/main.tf line 88, in resource "null_resource" "shell":
│   88:   provisioner "local-exec" {
│ 
│ Error running command 'bash
│ /builds/platforms/repositoryname/environments/prod/.terraform/modules/il_module/run.sh
│ L2J1aWxkcy9wbGF0Zm9ybXMvcmVwb3NpdG9yeW5hbWUvZW52aXJvbm1lbnRzL3Byb2QvLnRlcnJhZm9ybS9tb2R1bGVzL2lsX21vZHVsZS90bXBmaWxlcw==
│ 19816ec5-e5b8-d70a-b9e0-7c3ae0af6e12 1 0 1
│ YmFzZTY0LWVuY29kZWQtcmVkYWN0ZWQtY3VybC1jb21tYW5k
│ 19816ec5-e5b8-d70a-b9e0-7c3ae0af6e12.stdout
│ 19816ec5-e5b8-d70a-b9e0-7c3ae0af6e12.stderr
│ 19816ec5-e5b8-d70a-b9e0-7c3ae0af6e12.exitcode 0 0 0': exit status 127.
│ Output: /bin/sh: bash: not found
│ 
╵
```

It would be great if compatibility with Alpine and the official Hashicorp container environments could be restored.

**Comment by KyleKotowick** [comment]
Thanks for the info. Could you please try with experimental version `v0.0.1` and see if it works for your case? If so, I'll push it as a main release.

**Comment by jantari** [comment]
Hi, I had a chance to test it today - it's getting further, but now it's blocking at this issue: https://github.com/Invicton-Labs/terraform-external-shell-resource/pull/3

Output from terraform apply in a test project below:

<details>

```
module.randomword_server[0].random_uuid.uuid: Creating...
module.randomword_server[0].random_uuid.uuid: Creation complete after 0s [id=7ad7f03f-259a-c679-d52d-f44af65a4a86]
module.randomword_server[0].null_resource.shell: Creating...
module.randomword_server[0].null_resource.shell: Provisioning with 'local-exec'...
module.randomword_server[0].null_resource.shell (local-exec): Executing: ["/bin/sh" "-c" "/bin/sh /builds/platforms/testing-iv-sr-001-fix/environments/test/.terraform/modules/randomword_server/run.sh L2J1aWxkcy9wbGF0Zm9ybXMvdGVzdGluZy1pdi1zci0wMDEtZml4L2Vudmlyb25tZW50cy90ZXN0Ly50ZXJyYWZvcm0vbW9kdWxlcy9yYW5kb213b3JkX3NlcnZlci90bXBmaWxlcw== 7ad7f03f-259a-c679-d52d-f44af65a4a86 1 0 1 Y3VybCAtWCBQT1NUIC1rcyAtSCAnQWNjZXB0OiB0ZXh0L3BsYWluJyBodHRwczovL2ludGVybmFsYXBpL2FwaS9lbnRpdHk= 7ad7f03f-259a-c679-d52d-f44af65a4a86.stdout 7ad7f03f-259a-c679-d52d-f44af65a4a86.stderr 7ad7f03f-259a-c679-d52d-f44af65a4a86.exitcode 0 0 0"]
module.randomword_server[0].null_resource.shell (local-exec): base64: unrecognized option: decode
module.randomword_server[0].null_resource.shell (local-exec): BusyBox v1.34.1 (2022-04-04 10:19:27 UTC) multi-call binary.
module.randomword_server[0].null_resource.shell (local-exec): Usage: base64 [-d] [-w COL] [FILE]
module.randomword_server[0].null_resource.shell (local-exec): Base64 encode or decode FILE to standard output
module.randomword_server[0].null_resource.shell (local-exec): 	-d	Decode data
module.randomword_server[0].null_resource.shell (local-exec): 	-w COL	Wrap lines at COL (default 76, 0 disables)
╷
│ Error: local-exec provisioner error
│ 
│   with module.randomword_server[0].null_resource.shell,
│   on .terraform/modules/randomword_server/main.tf line 88, in resource "null_resource" "shell":
│   88:   provisioner "local-exec" {
│ 
│ Error running command '/bin/sh
│ /builds/platforms/testing-iv-sr-001-fix/environments/test/.terraform/modules/randomword_server/run.sh
│ L2J1aWxkcy9wbGF0Zm9ybXMvdGVzdGluZy1pdi1zci0wMDEtZml4L2Vudmlyb25tZW50cy90ZXN0Ly50ZXJyYWZvcm0vbW9kdWxlcy9yYW5kb213b3JkX3NlcnZlci90bXBmaWxlcw==
│ 7ad7f03f-259a-c679-d52d-f44af65a4a86 1 0 1
│ Y3VybCAtWCBQT1NUIC1rcyAtSCAnQWNjZXB0OiB0ZXh0L3BsYWluJyBodHRwczovL2ludGVybmFsYXBpL2FwaS9lbnRpdHk=
│ 7ad7f03f-259a-c679-d52d-f44af65a4a86.stdout
│ 7ad7f03f-259a-c679-d52d-f44af65a4a86.stderr
│ 7ad7f03f-259a-c679-d52d-f44af65a4a86.exitcode 0 0 0': exit status 1.
│ Output: base64: unrecognized option: decode
│ BusyBox v1.34.1 (2022-04-04 10:19:27 UTC) multi-call binary.
│ 
│ Usage: base64 [-d] [-w COL] [FILE]
│ 
│ Base64 encode or decode FILE to standard output
│ 
│ 	-d	Decode data
│ 	-w COL	Wrap lines at COL (default 76, 0 disables)
│ 
╵
Cleaning up project directory and file based variables
00:00
ERROR: Job failed: exit code 1
```
</details>

**Comment by KyleKotowick** [comment]
This issue has been resolved in release v0.4.0. Please try it out and report any problems as new issues.
